Week1/js3-hw-w1/app.js

```diff
@@ -0,0 +1,116 @@
+'user strict';
+
+
+const url = "https://api.github.com/orgs/HackYourFuture/repos?per_page=100";
+
+function fetchJSON(url, cb) {
+
+    const request = new XMLHttpRequest();
+
+    request.open("GET", url);
+    request.responseType = "json";
+
+    request.onreadystatechange = () => {
+        if (request.readyState === 4) {
+            if (request.status < 400) {
+                cb(null, request.response);
+            } else {
+                cb(new Error(request.statusText));
+            }
+        }
+    };
+
+    request.send();
+}
+
+
+function callback(error, data) {
+    if (error !== null) {
+    } else {
+        renderRepo(data);
+    }
+}
+fetchJSON(url, callback);
+
+function renderRepo(repo) {
+    const root = document.getElementById('root');
+    const listItem = createAndAppend('div', root);
+    listItem.id = 'listItem';
+    const listItemName = createAndAppend('p', listItem);
+    listItemName.innerHTML = 'HackYourFuture Repositories';
+    const select = createAndAppend('select', listItem);
+    const repoBox = createAndAppend('div', root);
+    repoBox.id = 'repo-box';
+    const contributorBox = createAndAppend('div', root);
+    contributorBox.id = 'contributors-box';
+    select.addEventListener("change", () => repoInfo(select.value));
+    repo.forEach(repo => {
+        const option = createAndAppend('option', select);
+        option.innerHTML = repo.name;
+        option.setAttribute('value', repo.name);
+    });
+    repoInfo(select.value);
+}
+
+function createAndAppend(tagName, parent) {
+    const element = document.createElement(tagName);
+    parent.appendChild(element);
+    return element;
+}
+
+function repoInfo(repoName) {
+    const urlContributor = 'https://api.github.com/repos/HackYourFuture/' + repoName + '/contributors';
+    const repoUrl = 'https://api.github.com/repos/HackYourFuture/' + repoName;
```

Comment:
You should not put static links inside functions, you should save these strings in a variable and then get the value from the variable. The way you did it gets the work done, but it will cause problems when you try to change the link (and forget to change it on some places).